goldstone/core/models.py

```diff
@@ -2634,3 +2636,70 @@ def integration(cls):
     def resourcetype(cls):
 
         return "routers"
+
+
+class DefinedSearch(models.Model):
+    """Defined searches, both system and user-created."""
+
+    # This object's Goldstone UUID.
+    uuid = UUIDField(version=1, auto=True, primary_key=True)
+
+    name = models.CharField(max_length=64)
+
+    description = models.CharField(max_length=1024, default='Defined Search')
+
+    query = models.CharField(max_length=512,
+                             help_text='YAML Elasticsearch query body')
+
+    protected = models.BooleanField(default=False,
+                                    help_text='True if this is system-defined')
+
+    created = CreationDateTimeField(editable=False, blank=True)
+    updated = ModificationDateTimeField(editable=True, blank=True)
+
+    @classmethod
+    def field_has_raw(cls, _):
+        """Bypass the "field has raw representation" check in
+        drfes.models.DailyIndexDocType."""
+
+        return False
+
+    @classmethod
+    def execute_query(cls, name):
+        """ executes an elasticsearch query and returns the result of that query.
+        :param query:
+        :return:
+        """
+
+        query = ast.literal_eval(DefinedSearch.objects.get(name=name))
+        # TBD : call bounded_search with timestamps once you add them
+        queryset = DailyIndexDocType.search(cls)
+        queryset = queryset.query(Q(query))
+
+        return queryset
+
+    def __unicode__(self):
+        """Return a useful string."""
+
+        return u'defined search UUID %s' % self.uuid
+
+    class Meta:               # pylint: disable=C0111,W0232,C1001
+        verbose_name_plural = "defined searches"
+
+
+class EventQueryDef(DefinedSearch):
+
+    """ Defines an event-type based elastic search query definition """
+
+    @classmethod
+    def search_by_syslog_err_type(cls):
+        """Return event rows that match the passed-in syslog error."""
+        result = DefinedSearch.execute_query('sys_log_err_type_search_1')
```

Comment:
After I added admin.register(DefinedSearch), py27 failed with NameError :
'DefinedSearch" not found.

I think the tests dont pull the models automatically in the background, I
probably need to add an explicit line in all the test files like this :

from .models import PolyResource, DefinedSearch, etc etc



On Thu, Dec 10, 2015 at 8:44 AM, John DeRosa <notifications@github.com>
wrote:

> In goldstone/core/models.py
> <https://github.com/Solinea/goldstone-server/pull/156#discussion_r47250892>
> :
>
> > +        """Return a useful string."""
> > +
> > +        return u'defined search UUID %s' % self.uuid
> > +
> > +    class Meta:               # pylint: disable=C0111,W0232,C1001
> > +        verbose_name_plural = "defined searches"
> > +
> > +
> > +class EventQueryDef(DefinedSearch):
> > +
> > +    """ Defines an event-type based elastic search query definition """
> > +
> > +    @classmethod
> > +    def search_by_syslog_err_type(cls):
> > +        """Return event rows that match the passed-in syslog error."""
> > +        result = DefinedSearch.execute_query('sys_log_err_type_search_1')
>
> Add a blank line after the docstring.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Solinea/goldstone-server/pull/156/files#r47250892>.
>